Rewrite 2020

promise opened the pull request:
See issue #5 for more information on this.

promise committed:
Starting from scratch

promise committed:
Set up the repository

promise committed:
Core bot files

promise committed:
Require all commands on startup

promise committed:
Optimization (doesn't run multiple functions)

promise committed:
Typos

promise committed:
Use async/await in db.guild

promise committed:
Have optional values to the function

promise committed:
Help and Info-commands, both untested

promise committed:
Typo

promise committed:
Update help command

promise committed:
Update info command

promise committed:
Add guild db as an argument, and added global db to optional list

promise committed:
Add another permission level, for server owners

promise committed:
Create factory reset command

promise committed:
Switch from Twitter integration to IFTTT/webhook integration

promise committed:
Bug fixes and optimization

promise committed:
Fixed DeepScan issue, timedout variable would always be false

promise committed:
Fixed Restyled.io issues

promise committed:
Disabled renovate.json, as it's inaccurate and crashing from semicolons

promise committed:
Create dump command

promise committed:
Change examples to objects instead of arrays

promise committed:
Modify help command tineout from two minutes to one minute

promise committed:
Create eval command

promise committed:
Add console logging for database errors

promise committed:
Create addpin command, needs testing

promise committed:
Create addregex command, needs testing

promise committed:
Create addrole command, needs testing

promise committed:
Bug fixes

promise committed:
Add invite link to info-command, may change in the future

promise committed:
Formatted checkArgs better, and added note to permissions

promise committed:
Improvements

promise committed:
Made arguments have a constant message instead

promise committed:
Wrong command, my bad.

promise committed:
Made the base64-encoder an actual base64-based encoder

promise committed:
Removed unused arg

promise committed:
Autosetup command

promise committed:
Changed invalid arguemnts message

promise committed:
Editpin command

promise committed:
Change some error messages

promise committed:
Changed command handler arguments

promise committed:
Removed mode "score" from editpin as this is not a functionality

promise committed:
Editrole command

promise committed:
Link command

promise committed:
Listregex command

promise committed:
Listroles command

promise committed:
Add documentation

promise committed:
Add start script

promise committed:
Last part of editrole command

promise committed:
Changed some messages

promise committed:
Create liveboard command (for non-premium)

promise committed:
Module command

promise committed:
Create modifications command

promise committed:
Import generateID function

promise committed:
Notifyme command

promise committed:
Added emojis in embeds

promise committed:
Ping command

promise committed:
Prefix command

promise committed:
Bug fixes

promise committed:
DeepScan issues

promise committed:
Add config.example.json

promise committed:
Documentation update

promise committed:
Added more and formatted aliases

promise committed:
Typos

promise committed:
Bug fixes

promise committed:
Added search query limitations for help command

promise committed:
Renamed module-command to toggle-command

promise committed:
More bug fixes

promise committed:
Scripts now has indents

promise committed:
Add more debug info

promise committed:
Add sharding availability

promise committed:
Update command documentation

promise committed:
Create invite command

promise committed:
Change links in info command

promise committed:
Update module descriptions

promise committed:
Update autodoc to add an explanation of how to use commands

promise committed:
Add features to documentation

promise committed:
Update selfhosting command

promise committed:
Add links to documentation

promise committed:
DeepScan fix(es)

Shompi commented on commands/addpin.js:
```diff
@@ -1,32 +1,33 @@
+  let { pins: alreadyGeneratedPins } = await gdb.get(), id = generateID(Object.keys(alreadyGeneratedPins))
+  gdb.setPin(id, mode, count, action)
+    .then(() => message.channel.send("✅ Pintrigger with ID `" + id + "` now saved."))
+    .catch(e => console.log(e) && message.channel.send("🆘 An unknown database error occoured. Please try again, or contact support."))
```
```diff
-occoured
+occurred
```

Shompi commented on commands/addregex.js:
```diff
@@ -1,15 +1,31 @@
-        .catch(err => { console.log(err); botMsg.edit("❌ " + strings["UNKNOWN_ERROR"]) });
+  gdb.addRegex(regex)
+    .then(() => message.channel.send("✅ Regex `" + regex + "` is now saved."))
+    .catch(e => console.log(e) && message.channel.send("🆘 An unknown database error occoured. Please try again, or contact support."))
```
```diff
- occoured
+ occurred
```

Shompi commented on commands/addrole.js:
```diff
@@ -1,37 +1,41 @@
+  let { roles: alreadyGeneratedPins } = await gdb.get(), id = generateID(Object.keys(alreadyGeneratedPins))
+  gdb.setRole(id, mode, count, duration)
+    .then(() => message.channel.send("✅ Rolereward with ID `" + id + "` is now saved."))
+    .catch(e => console.log(e) && message.channel.send("🆘 An unknown database error occoured. Please try again, or contact support."))
```
```diff
- occoured
+ occurred
```

promise committed:
Typos

promise committed:
Performance boosts

promise committed:
Error handling

promise committed:
More stats in info command. Ping removed.

promise committed:
Redesigned ping command

promise committed:
Small performance boosts

promise committed:
Bug fixes

promise committed:
Bug fixing

promise committed:
Dependencies

promise committed:
Import scores command

promise committed:
Export scores command

promise committed:
Listpins command

promise committed:
Reset score command

promise committed:
Changed some messages

promise committed:
Changed some commands

promise committed:
setMultiple-function and bug fixing

promise committed:
Unlink command

promise committed:
Top command

promise committed:
Settopic command

promise committed:
Setscore command

promise committed:
Renamed prefix command

promise committed:
Add embed error message

promise committed:
Setcount command

promise committed:
Command description changes

promise committed:
Changed timeout duration from one minute to three minutes

promise committed:
Resetcount command

promise committed:
Changed resetscore command

promise committed:
Update commands

promise committed:
Description edits

promise committed:
Changed all "for (var.." to "for (const.."

promise committed:
Changed const to let

promise committed:
Remove logging commands

promise committed:
Renamed top command

promise committed:
Generated favicon

promise committed:
Changed alias system

ImgBotApp committed:
[ImgBot] Optimize images

/docs/icon.png -- 163.61kb -> 144.86kb (11.46%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

promise committed:
Merge pull request #7 from gleeny/imgbot

[ImgBot] Optimize images

promise committed:
Changed vars to lets

promise committed:
Bug fixing

promise committed:
Performance

promise committed:
Documentation

promise committed:
Unused code

promise committed:
Made commands be in codeblock

promise committed:
Typos+

promise committed:
Add filesize limit (will support ~1300 members at least.)

promise committed:
Removenotif command

promise committed:
Removepin command

promise committed:
Removeregex command

promise committed:
Removerole command

promise committed:
Removetimeoutrole command

promise committed:
Settimeoutrole command

promise committed:
Merge branch 'rewrite-2020' of https://github.com/gleeny/countr into rewrite-2020

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 57.28kb -> 54.36kb (5.09%)

/docs/icons/favicon/mstile-144x144.png -- 1.57kb -> 1.27kb (19.44%)
/docs/icons/favicon/mstile-150x150.png -- 1.66kb -> 1.38kb (16.84%)
/docs/icons/favicon/apple-touch-icon-120x120.png -- 2.89kb -> 2.64kb (8.72%)
/docs/icons/favicon/safari-pinned-tab.svg -- 3.20kb -> 3.01kb (5.74%)
/docs/icons/favicon/apple-touch-icon-180x180.png -- 4.61kb -> 4.37kb (5.22%)
/docs/icons/favicon/apple-touch-icon.png -- 4.61kb -> 4.37kb (5.22%)
/docs/icons/favicon/apple-touch-icon-144x144.png -- 3.04kb -> 2.89kb (5.08%)
/docs/icons/favicon/apple-touch-icon-152x152.png -- 3.63kb -> 3.45kb (4.95%)
/docs/icons/favicon/android-chrome-512x512.png -- 20.43kb -> 19.55kb (4.29%)
/docs/icons/favicon/android-chrome-192x192.png -- 6.49kb -> 6.33kb (2.48%)
/docs/icons/favicon/apple-touch-icon-76x76.png -- 1.50kb -> 1.48kb (0.91%)
/docs/icons/favicon/apple-touch-icon-114x114.png -- 2.24kb -> 2.22kb (0.87%)
/docs/icons/favicon/apple-touch-icon-72x72.png -- 1.42kb -> 1.41kb (0.62%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

promise committed:
Merge pull request #8 from gleeny/imgbot

[ImgBot] Optimize images

promise committed:
DeepScan issue fix

promise committed:
Docs

promise committed:
Changed function

promise committed:
Bug fixing

promise committed:
Made so if the channel hits 50 pinned messages (limit), we unpin the oldest count.

promise committed:
Embed errors

promise committed:
Bug fixing

promise committed:
Reset prefix ability

promise committed:
Disable timeout role command

promise committed:
Add parent channel

promise committed:
Update sharding.js

promise committed:
Add more disabled events

promise committed:
Made it more obvious where on the scoreboard you are, plus bug fix

promise committed:
Removefromscore command

promise committed:
Addtoscore command

promise committed:
Bug fix, score is not supposed to go under 0.

promise committed:
Topic bug fixing

promise committed:
Changed it a bit

promise committed:
Update command infos

promise committed:
Update documentation

promise committed:
Recover module bug fixing

promise committed:
Bug fixing

promise committed:
Update getWeek function

promise committed:
Add activity logs

promise committed:
Docs

promise committed:
Docs

promise committed:
Liveboard command update

promise committed:
Issue templates

promise committed:
Switch from trello to gh-projects

promise committed:
Moved docs to branch gh-pages

promise committed:
Autodoc.js relocated

promise committed:
Dates issue fixed

fabianneufang committed:
Create webboard.js

fabianneufang committed:
Update webboard.js

promise committed:
Update webboard.js

promise committed:
Merge pull request #9 from gleeny/thefabicraft-github-patch-1

Web interface

promise committed:
Bug fixing with timeout roles upon startup

promise committed:
Liveboard usage update

promise committed:
Bug fixes

promise committed:
Prevent spam in the console when using reactions
